src/Aquifer.Jobs/Subscribers/TranslationMessageSubscriber.cs

```diff
@@ -393,12 +393,28 @@ private async Task TranslateResourceCoreAsync(int resourceContentId,
 
         resourceContentVersion.ResourceContent.ContentUpdated = DateTime.UtcNow;
 
-        // Save a snapshot of the translation.
+        // If not a retranslation then save a new snapshot of the translation.
         // Always save it as TranslationAwaitingAiDraft in order to prevent a user's name from appearing with the snapshot.
-        await _resourceHistoryService.AddSnapshotHistoryAsync(resourceContentVersion,
-            startedByUserId,
-            ResourceContentStatus.TranslationAwaitingAiDraft,
-            ct);
+        if (!shouldForceRetranslation)
+        {
+            await _resourceHistoryService.AddSnapshotHistoryAsync(resourceContentVersion,
+                startedByUserId,
+                ResourceContentStatus.TranslationAwaitingAiDraft,
+                ct);
+        }
+        // If a retranslation then update the existing snapshot to not create too many snapshots.
+        else
+        {
+            var existingAiTranslationSnapshot = resourceContentVersion.ResourceContentVersionSnapshots
```

Comment:
Do we need to handle if this is null or will `Last` blow up if it can't find one?

Reply:
It will blow up if it can't find one.  If that happens we want it to blow up because a dev put a bad retranslate message on the queue but we could admittedly have a better error message.